frame/treasury/src/lib.rs

```diff
@@ -221,6 +269,25 @@ decl_storage! {
 		/// Simple preimage lookup from the reason's hash to the original data. Again, has an
 		/// insecure enumerable hash since the key is guaranteed to be the result of a secure hash.
 		pub Reasons get(fn reasons): map hasher(identity) T::Hash => Option<Vec<u8>>;
+
+		/// Number of bounty proposals that have been made.
+		pub BountyCount get(fn bounty_count): BountyIndex;
+
+		/// Bounties that have been made.
+		pub Bounties get(fn bounties):
+			map hasher(twox_64_concat) BountyIndex
+			=> Option<Bounty<T::AccountId, BalanceOf<T>>>;
+
+		/// The status of each bounty.
+		pub BountyStatuses get(fn bounty_statuses):
+			map hasher(twox_64_concat) BountyIndex => Option<BountyStatus>;
+
+		/// The bounty beneficiary and the block the fund can be claimed
+		pub BountyBeneficiary get(fn bounty_beneficiary):
+			map hasher(twox_64_concat) BountyIndex => Option<(T::AccountId, T::BlockNumber)>;
+
+		/// Bounty indices that have been approved but not yet funded.
+		pub BountyApprovals get(fn bounty_approvals): Vec<BountyIndex>;
```

Comment:
Is it sorted? Vecs are bad. Store in one of the previous three map items as an extra field.

Reply:
This is kind of a task queue, I basically copied
https://github.com/paritytech/substrate/blob/4269e2308eca48a90e724890c900f0421b667fcc/frame/treasury/src/lib.rs#L214-L215

They are batch processed at end of a spending period in approval order, just like how treasury proposal is processed.